### ISSUE TITLE
RFC: Add `convert_to_generic_platform` function to wheeltools module

### ISSUE THREAD

**Issue by jcfr** [anthrotype/wheeltools#2]
When creating wheels packaging executable and tools that (1) are compiled  and (2)  do not have any dependency on the python ABI, it is convenient to update the wheel to capture that.

Doing so avoid to compile a set of wheels for each python release.

For example, these wheels

```
awesometool-1.2.3-cp27-cp27m-manylinux1_i686.whl 
awesometool-1.2.3-cp27-cp27mu-manylinux1_x86_64.whl 
awesometool-1.2.3-cp34-cp34m-macosx_10_6_x86_64.whl
awesometool-1.2.3-cp36-cp36m-win_amd64.whl
```

can be updated to:

```
awesometool-1.2.3-py2-none-manylinux1_i686.whl 
awesometool-1.2.3-py2-none-manylinux1_x86_64.whl 
awesometool-1.2.3-py3-none-macosx_10_6_x86_64.whl
awesometool-1.2.3-py3-none-win_amd64.whl
```

In the context of the [cmake-python-distributions](https://github.com/scikit-build/cmake-python-distributions), we already wrote a CLI (that depends on wheeltools) to perform this transformation.

See [convert_to_generic_platform_wheel.py](https://github.com/scikit-build/cmake-python-distributions/blob/d74ee54eb21d989b345ec967f2aab73700048162/scripts/convert_to_generic_platform_wheel.py)

Since we would like to easily re-use the corresponding functionality in other projects (e.g [ninja-python-distributions](https://github.com/scikit-build/ninja-python-distributions)), it would be nice to contribute the functionality to this project.

@anthrotype  What do you think ?

**Comment by jcfr** [comment]
Following discussion with @anthrotype, the project has been transferred to the scikit-build organization. See [here](https://github.com/scikit-build/wheeltools/pull/1#issuecomment-496443774) for more details about this.

**Comment by anthrotype** [comment]
Yes, I'd like that!
I use similar hacks in another project of mine where I embed a shared library (that in turn does not link with cpython) as package_data and wrap it with ctypes.
I won't be able to work on this feature though.

**Comment by jcfr** [comment]
Thanks for the feedback.

I should be able to move this forward in the next few weeks. In the mean time, does the name `convert_to_generic_platform` make sense ?

**Comment by anthrotype** [comment]
to me, that sounds a bit ambiguous because the "platform" (e.g. `manylinux1_x86_64` or `win_amd64`) is still very specific; what's generic is the (what I belive they call) `abi` tag in the middle which is set to `none` to signal that the wheel doesn't have any implementation-specific ABI requirements.

**Comment by jcfr** [comment]
Thanks for the feedback. Makes sense.

What about : 

`strip_abi_tags_from_wheel` 

or

`strip_abi_tags`

**Comment by anthrotype** [comment]
yeah, both of the latter work better than `convert_to_generic_platform`.
Thanks

**Comment by anthrotype** [comment]
came across this possibly related upstream wheel issue: https://github.com/pypa/wheel/issues/175
